core-web/libs/portlets/edit-ema/portlet/src/lib/dot-ema-shell/dot-ema-shell.component.spec.ts

```diff
@@ -643,6 +684,246 @@ describe('DotEmaShellComponent', () => {
             });
         });
 
+        describe('language checking', () => {
+            it('should not trigger the confirmation service if the page is translated to the current language', () => {
+                const confirmationServiceSpy = jest.spyOn(confirmationService, 'confirm');
+
+                spectator.detectChanges();
+
+                expect(confirmationServiceSpy).not.toHaveBeenCalled();
+            });
+
+            it("should trigger the confirmation service if the page isn't translated to the current language", fakeAsync(() => {
+                const confirmationServiceSpy = jest.spyOn(confirmationService, 'confirm');
+                spectator.triggerNavigation({
+                    url: [],
+                    queryParams: {
+                        language_id: 2,
+                        url: 'index',
+                        'com.dotmarketing.persona.id': DEFAULT_PERSONA.identifier
+                    }
+                });
+
+                spectator.detectChanges();
+
+                tick();
+
+                expect(confirmationServiceSpy).toHaveBeenCalledWith({
+                    accept: expect.any(Function),
+                    acceptEvent: expect.any(Object),
+                    reject: expect.any(Function),
+                    rejectEvent: expect.any(Object),
+                    rejectIcon: 'hidden',
+                    acceptIcon: 'hidden',
+                    key: 'shell-confirm-dialog',
+                    header: 'editpage.language-change-missing-lang-populate.confirm.header',
+                    message: 'editpage.language-change-missing-lang-populate.confirm.message'
+                });
+            }));
+
+            it('should trigger a navigation to default language when the user rejects the creation', fakeAsync(() => {
+                spectator.triggerNavigation({
+                    url: [],
+                    queryParams: {
+                        language_id: 2,
+                        url: 'index',
+                        'com.dotmarketing.persona.id': DEFAULT_PERSONA.identifier
+                    }
+                });
+
+                spectator.detectChanges();
+
+                tick(1000);
+                spectator.detectChanges();
+
+                const confirmDialog = spectator.query(byTestId('confirm-dialog'));
+
+                confirmDialog
+                    .querySelector('.p-confirm-dialog-reject')
+                    .dispatchEvent(new MouseEvent('click'));
```

Comment:
@zJaaal I suggest using `createMouseEvent('click');` by the spectator to create mock events.